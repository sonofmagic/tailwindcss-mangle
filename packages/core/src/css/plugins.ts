import type { PluginCreator } from 'postcss'
import defu from 'defu'
import parser from 'postcss-selector-parser'
import { IClassGeneratorContextItem, ICssHandlerOptions } from '@/types'
export type PostcssMangleTailwindcssPlugin = PluginCreator<ICssHandlerOptions>

const postcssPlugin = 'postcss-mangle-tailwindcss-plugin'

export function isVueScoped(s: parser.ClassName): boolean {
  if (s.parent) {
    const index = s.parent.nodes.indexOf(s)
    if (index > -1) {
      const nextNode = s.parent.nodes[index + 1]
      if (nextNode && nextNode.type === 'attribute' && nextNode.attribute.includes('data-v-')) {
        return true
      }
    }
  }
  return false
}

const postcssMangleTailwindcssPlugin: PostcssMangleTailwindcssPlugin = (options) => {
  const { ignoreVueScoped } = defu(options, {
    ignoreVueScoped: true
  })
  if (options?.scene === 'loader') {
    // must set newClassMap options
    let set: Set<string> = new Set()

    if (options && options.runtimeSet) {
      set = options.runtimeSet
    }
    return {
      postcssPlugin,
      Rule(rule) {
        rule.selector = parser((selectors) => {
          selectors.walkClasses((s) => {
            if (s.value) {
              const existed = set.has(s.value)

              if (existed) {
                if (ignoreVueScoped && isVueScoped(s)) {
                  return
                }
                s.value = options.classGenerator.generateClassName(s.value).name
              }
            }
          })
        }).processSync(rule.selector)
      }
    }
  } else {
    let newClassMap: Record<string, IClassGeneratorContextItem> = {}

    if (options && options.classGenerator) {
      newClassMap = options.classGenerator.newClassMap
    }

    return {
      postcssPlugin,
      Rule(rule) {
        rule.selector = parser((selectors) => {
          selectors.walkClasses((s) => {
            if (s.value) {
              const hit = newClassMap[s.value]
              const existed = Boolean(hit)

              if (existed) {
                if (ignoreVueScoped && isVueScoped(s)) {
                  return
                }
                s.value = hit.name
              }
            }
          })
        }).processSync(rule.selector)
      }
    }
  }
}

postcssMangleTailwindcssPlugin.postcss = true

export { postcssMangleTailwindcssPlugin }
