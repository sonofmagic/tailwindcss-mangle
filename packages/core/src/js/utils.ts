import type babel from '@babel/core'
// const t = babel.types
export function getStringLiteralCalleeName(path: babel.NodePath<babel.types.StringLiteral>) {
  if (path.parentPath.isCallExpression()) {
    const callee = path.parentPath.get('callee')
    if (callee.isIdentifier()) {
      return callee.node.name
    }
  }
}

export function getTemplateElementCalleeName(path: babel.NodePath<babel.types.TemplateElement>) {
  if (path.parentPath.isTemplateLiteral()) {
    const pp = path.parentPath
    if (pp.parentPath.isCallExpression()) {
      const callee = pp.parentPath.get('callee')
      if (callee.isIdentifier()) {
        return callee.node.name
      }
    }
  }
}
