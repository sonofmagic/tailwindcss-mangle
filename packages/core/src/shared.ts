export * from '@tailwindcss-mangle/shared'
