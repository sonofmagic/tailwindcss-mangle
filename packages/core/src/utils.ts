import micromatch from 'micromatch'

const { isMatch } = micromatch

export function escapeStringRegexp(str: string) {
  if (typeof str !== 'string') {
    throw new TypeError('Expected a string')
  }
  return str.replaceAll(/[$()*+.?[\\\]^{|}]/g, '\\$&').replaceAll('-', '\\x2d')
}

export function createGlobMatcher(pattern: string | string[] | undefined, fallbackValue: boolean = false) {
  if (pattern === undefined) {
    return function () {
      return fallbackValue
    }
  }
  return function (file: string) {
    return isMatch(file, pattern)
  }
}

export { defaultMangleClassFilter, isMap, isRegexp } from '@tailwindcss-mangle/shared'
