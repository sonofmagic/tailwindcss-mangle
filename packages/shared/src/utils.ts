export const preserveClassNames = [
  // https://tailwindcss.com/docs/transition-timing-function start
  // https://github.com/sonofmagic/tailwindcss-mangle/issues/21
  'ease-out',
  'ease-linear',
  'ease-in',
  'ease-in-out'
  // https://tailwindcss.com/docs/transition-timing-function end
]

// eslint-disable-next-line unicorn/no-array-reduce
const preserveClassNamesMap = preserveClassNames.reduce<Record<(typeof preserveClassNames)[number], true>>((acc, cur) => {
  acc[cur] = true
  return acc
}, {})

export const defaultMangleClassFilter = (className: string) => {
  if (preserveClassNamesMap[className]) {
    return false
  }
  // ignore className like 'filter','container'
  // it may be dangerous to mangle/rename all StringLiteral , so use /-/ test for only those with /-/ like:
  // bg-[#123456] w-1 etc...
  return /[:-]/.test(className)
}

export function groupBy<T>(arr: T[], cb: (arg: T) => string): Record<string, T[]> {
  if (!Array.isArray(arr)) {
    throw new TypeError('expected an array for first argument')
  }

  if (typeof cb !== 'function') {
    throw new TypeError('expected a function for second argument')
  }

  const result: Record<string, T[]> = {}
  for (const item of arr) {
    const bucketCategory = cb(item)
    const bucket = result[bucketCategory]

    if (Array.isArray(bucket)) {
      result[bucketCategory].push(item)
    } else {
      result[bucketCategory] = [item]
    }
  }

  return result
}

export const acceptChars = [...'abcdefghijklmnopqrstuvwxyz']

export function stripEscapeSequence(words: string) {
  return words.replaceAll('\\', '')
}

export function isRegexp(value: unknown) {
  return Object.prototype.toString.call(value) === '[object RegExp]'
}

export function isMap(value: unknown) {
  return Object.prototype.toString.call(value) === '[object Map]'
}

export function regExpTest(arr: (string | RegExp)[] = [], str: string) {
  if (Array.isArray(arr)) {
    for (const item of arr) {
      if (typeof item === 'string') {
        if (item === str) {
          return true
        }
      } else if (isRegexp(item)) {
        item.lastIndex = 0
        if (item.test(str)) {
          return true
        }
      }
    }
    return false
  }
  throw new TypeError("paramater 'arr' should be a Array of Regexp | String !")
}
