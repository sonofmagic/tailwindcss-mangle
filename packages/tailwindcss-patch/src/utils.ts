import fss from 'node:fs'
import fs from 'node:fs/promises'
import type { SyncOpts } from 'resolve'
import pkg from 'resolve'

const { sync } = pkg

export function ensureFileContent(filepaths: string | string[]) {
  if (typeof filepaths === 'string') {
    filepaths = [filepaths]
  }
  let content
  for (const filepath of filepaths) {
    if (fss.existsSync(filepath)) {
      content = fss.readFileSync(filepath, {
        encoding: 'utf8',
      })
      break
    }
  }
  return content
}

export function requireResolve(id: string, opts?: SyncOpts) {
  return sync(id, opts)
}

export async function ensureDir(p: string) {
  try {
    await fs.access(p)
  }
  catch {
    await fs.mkdir(p, {
      recursive: true,
    })
  }
}
