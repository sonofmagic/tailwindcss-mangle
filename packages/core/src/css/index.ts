import postcss from 'postcss'
import { ICssHandlerOptions } from '../types'
import { postcssMangleTailwindcssPlugin } from './plugins'

export function cssHandler(rawSource: string, options: ICssHandlerOptions) {
  const acceptedPlugins = [postcssMangleTailwindcssPlugin(options)]
  return postcss(acceptedPlugins).process(rawSource).css
}
