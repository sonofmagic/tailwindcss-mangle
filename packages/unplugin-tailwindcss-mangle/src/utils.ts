import fs from 'node:fs'
import path from 'node:path'
import micromatch from 'micromatch'

import { groupBy } from '@tailwindcss-mangle/shared'
import { pluginName } from './constants'
const { isMatch } = micromatch

export function getGroupedEntries<T>(
  entries: [string, T][],
  options = {
    cssMatcher(file: string) {
      return /\.css$/.test(file)
    },
    htmlMatcher(file: string) {
      return /\.html?$/.test(file)
    },
    jsMatcher(file: string) {
      return /\.[cm]?js$/.test(file)
    }
  }
) {
  const { cssMatcher, htmlMatcher, jsMatcher } = options
  const groupedEntries = groupBy(entries, ([file]) => {
    if (cssMatcher(file)) {
      return 'css'
    } else if (htmlMatcher(file)) {
      return 'html'
    } else if (jsMatcher(file)) {
      return 'js'
    } else {
      return 'other'
    }
  })
  if (!groupedEntries.css) {
    groupedEntries.css = []
  }
  if (!groupedEntries.html) {
    groupedEntries.html = []
  }
  if (!groupedEntries.js) {
    groupedEntries.js = []
  }
  if (!groupedEntries.other) {
    groupedEntries.other = []
  }
  return groupedEntries as Record<'css' | 'html' | 'js' | 'other', [string, T][]>
}

export function createGlobMatcher(pattern: string | string[] | undefined, fallbackValue: boolean = false) {
  if (pattern === undefined) {
    return function () {
      return fallbackValue
    }
  }
  return function (file: string) {
    return isMatch(file, pattern)
  }
}

export function getCacheDir(basedir = process.cwd()) {
  return path.resolve(basedir, 'node_modules/.cache', pluginName)
}

export function mkCacheDirectory(cwd = process.cwd()) {
  const cacheDirectory = getCacheDir(cwd)

  const exists = fs.existsSync(cacheDirectory)
  if (!exists) {
    fs.mkdirSync(cacheDirectory, {
      recursive: true
    })
  }
  return cacheDirectory
}

export function cacheDump(filename: string, data: any[] | Set<any>, basedir?: string) {
  try {
    const dir = mkCacheDirectory(basedir)
    fs.writeFileSync(path.resolve(dir, filename), JSON.stringify([...data], undefined, 2), 'utf8')
  } catch (error) {
    console.log(error)
  }
}

export { defaultMangleClassFilter, isMap, isRegexp } from '@tailwindcss-mangle/shared'
