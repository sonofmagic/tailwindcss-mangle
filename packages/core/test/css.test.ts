import { cssHandler } from '@/css'
import { ClassGenerator } from '@/shared'
import { getTestCase } from './utils'
describe('css', () => {
  let classGenerator: ClassGenerator
  beforeEach(() => {
    classGenerator = new ClassGenerator()
  })

  it('vue scoped .gap-y-4', () => {
    const runtimeSet = new Set<string>()
    runtimeSet.add('gap-y-4')
    classGenerator.generateClassName('gap-y-4')
    const testCase = `@media (min-width: 768px) {
      .gap-y-4 {
      }
    }`

    const css = cssHandler(testCase, {
      classGenerator,
      runtimeSet
    })
    expect(css).toMatchSnapshot()
  })

  it('vue scoped .gap-y-4[data-v-0f84999b]', () => {
    const runtimeSet = new Set<string>()
    runtimeSet.add('gap-y-4')
    classGenerator.generateClassName('gap-y-4')
    const testCase = `@media (min-width: 768px) {
      .gap-y-4[data-v-0f84999b] {
      }
    }`

    const css = cssHandler(testCase, {
      classGenerator,
      runtimeSet
    })
    expect(css).toMatchSnapshot()
  })

  it('vue scoped no ignore .gap-y-4[data-v-0f84999b]', () => {
    const runtimeSet = new Set<string>()
    runtimeSet.add('gap-y-4')
    classGenerator.generateClassName('gap-y-4')
    const testCase = `@media (min-width: 768px) {
      .gap-y-4[data-v-0f84999b] {
      }
    }`

    const css = cssHandler(testCase, {
      classGenerator,
      runtimeSet,
      ignoreVueScoped: false
    })
    expect(css).toMatchSnapshot()
  })

  it('common with scoped', () => {
    const runtimeSet = new Set<string>()
    runtimeSet.add('bg-white')
    classGenerator.generateClassName('bg-white')
    const testCase = `
    .bg-white[data-v-0f84999b] {
      --tw-bg-opacity: 1;
      background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
    }`

    const css = cssHandler(testCase, {
      classGenerator,
      runtimeSet
    })
    expect(css).toMatchSnapshot()
  })

  it('vue.scoped.css', () => {
    const runtimeSet: Set<string> = new Set(JSON.parse(getTestCase('nuxt-app-partial-class-set.json')))
    const testCase = getTestCase('vue.scoped.css')
    const css = cssHandler(testCase, {
      classGenerator,
      runtimeSet
    })
    expect(css).toMatchSnapshot()
  })
})
