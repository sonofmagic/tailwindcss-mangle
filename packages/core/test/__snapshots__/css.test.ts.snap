// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`css > common with scoped 1`] = `
"
    .bg-white[data-v-0f84999b] {
      --tw-bg-opacity: 1;
      background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
    }"
`;

exports[`css > vue scoped .gap-y-4 1`] = `
"@media (min-width: 768px) {
      .tw-a {
      }
    }"
`;

exports[`css > vue scoped .gap-y-4[data-v-0f84999b] 1`] = `
"@media (min-width: 768px) {
      .gap-y-4[data-v-0f84999b] {
      }
    }"
`;

exports[`css > vue scoped no ignore .gap-y-4[data-v-0f84999b] 1`] = `
"@media (min-width: 768px) {
      .tw-a[data-v-0f84999b] {
      }
    }"
`;

exports[`css > vue.scoped.css 1`] = `
".flex[data-v-0f84999b] {
  display: flex;
}
.grid[data-v-0f84999b] {
  display: grid;
}
.hidden[data-v-0f84999b] {
  display: none;
}
.tw-h[data-v-0f84999b] {
  flex-direction: column;
}
.flex-col-reverse[data-v-0f84999b] {
  flex-direction: column-reverse;
}
.place-content-center[data-v-0f84999b] {
  place-content: center;
}
.tw-f[data-v-0f84999b] {
  align-items: center;
}
.justify-end[data-v-0f84999b] {
  justify-content: flex-end;
}
.tw-g[data-v-0f84999b] {
  justify-content: center;
}
.flex-1[data-v-0f84999b] {
  flex: 1 1 0%;
}
.order-last[data-v-0f84999b] {
  order: 9999;
}
.tw-l[data-v-0f84999b] {
  font-family: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, Liberation Mono, Courier New, monospace;
}
.tw-mc[data-v-0f84999b] {
  font-weight: 600;
}
.tw-ga[data-v-0f84999b] {
  font-weight: 700;
}
.h-32[data-v-0f84999b] {
  height: 8rem;
}
.h-\\\\[70px\\\\][data-v-0f84999b] {
  height: 70px;
}
.text-4xl[data-v-0f84999b] {
  font-size: 2.25rem;
  line-height: 2.5rem;
}
.text-xl[data-v-0f84999b] {
  font-size: 1.25rem;
  line-height: 1.75rem;
}
.tw-m[data-v-0f84999b] {
  font-size: 0.875rem;
  line-height: 1.25rem;
}
.tw-lc[data-v-0f84999b] {
  font-size: 1.5rem;
  line-height: 2rem;
}
.mx-auto[data-v-0f84999b] {
  margin-left: auto;
  margin-right: auto;
}
.mb-2[data-v-0f84999b] {
  margin-bottom: 0.5rem;
}
.max-w-\\\\[960px\\\\][data-v-0f84999b] {
  max-width: 960px;
}
.tw-a[data-v-0f84999b] {
  min-height: 100vh;
}
.p-1[data-v-0f84999b] {
  padding: 0.25rem;
}
.px-4[data-v-0f84999b] {
  padding-left: 1rem;
  padding-right: 1rem;
}
.py-14[data-v-0f84999b] {
  padding-bottom: 3.5rem;
  padding-top: 3.5rem;
}
.py-6[data-v-0f84999b] {
  padding-bottom: 1.5rem;
  padding-top: 1.5rem;
}
.tw-dc[data-v-0f84999b] {
  padding-left: 1.25rem;
  padding-right: 1.25rem;
}
.pt-\\\\[58px\\\\][data-v-0f84999b] {
  padding-top: 58px;
}
.tw-t[data-v-0f84999b] {
  padding-bottom: 1.5rem;
}
.absolute[data-v-0f84999b] {
  position: absolute;
}
.relative[data-v-0f84999b] {
  position: relative;
}
.inset-y-0[data-v-0f84999b] {
  bottom: 0;
  top: 0;
}
.inset-x-0[data-v-0f84999b] {
  left: 0;
  right: 0;
}
.tw-b[data-v-0f84999b] {
  left: 0;
}
.right-0[data-v-0f84999b] {
  right: 0;
}
.-top-\\\\[58px\\\\][data-v-0f84999b] {
  top: -58px;
}
.-top-3[data-v-0f84999b] {
  top: -0.75rem;
}
.focus-visible\\\\:ring-2[data-v-0f84999b]:focus-visible {
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(2px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow), var(--tw-ring-shadow), var(--tw-shadow, 0 0 #0000);
}
.tw-c[data-v-0f84999b] {
  text-align: center;
}
.text-black[data-v-0f84999b] {
  --tw-text-opacity: 1;
  color: rgba(0, 0, 0, var(--tw-text-opacity));
}
.text-gray-700[data-v-0f84999b] {
  --tw-text-opacity: 1;
  color: rgba(66, 66, 66, var(--tw-text-opacity));
}
.hover\\\\:text-black[data-v-0f84999b]:hover {
  --tw-text-opacity: 1;
  color: rgba(0, 0, 0, var(--tw-text-opacity));
}
.antialiased[data-v-0f84999b] {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
.tw-e[data-v-0f84999b] {
  width: 100%;
}
.w-\\\\[20\\\\%\\\\][data-v-0f84999b] {
  width: 20%;
}
.tw-d[data-v-0f84999b] {
  z-index: 10;
}
.z-1[data-v-0f84999b] {
  z-index: 1;
}
.gap-6[data-v-0f84999b] {
  grid-gap: 1.5rem;
  gap: 1.5rem;
}
.gap-3[data-v-0f84999b] {
  grid-gap: 0.75rem;
  gap: 0.75rem;
}
.gap-x-4[data-v-0f84999b] {
  grid-column-gap: 1rem;
  -webkit-column-gap: 1rem;
  -moz-column-gap: 1rem;
  column-gap: 1rem;
}
.gap-y-16[data-v-0f84999b] {
  grid-row-gap: 4rem;
  -webkit-row-gap: 4rem;
  -moz-row-gap: 4rem;
  row-gap: 4rem;
}
.gap-y-4[data-v-0f84999b] {
  grid-row-gap: 1rem;
  -webkit-row-gap: 1rem;
  -moz-row-gap: 1rem;
  row-gap: 1rem;
}
.gap-y-2[data-v-0f84999b] {
  grid-row-gap: 0.5rem;
  -webkit-row-gap: 0.5rem;
  -moz-row-gap: 0.5rem;
  row-gap: 0.5rem;
}
.grid-cols-2[data-v-0f84999b] {
  grid-template-columns: repeat(2, minmax(0, 1fr));
}
.col-span-2[data-v-0f84999b] {
  -ms-grid-column-span: span 2 / span 2;
  grid-column: span 2 / span 2;
}
.row-span-2[data-v-0f84999b] {
  -ms-grid-row-span: span 2 / span 2;
  grid-row: span 2 / span 2;
}
.transition-opacity[data-v-0f84999b] {
  transition-duration: 0.15s;
  transition-property: opacity;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
}
.duration-300[data-v-0f84999b] {
  transition-duration: 0.3s;
}
@media (min-width: 640px) {
  .sm\\\\:flex-row[data-v-0f84999b] {
    flex-direction: row;
  }
  .sm\\\\:justify-between[data-v-0f84999b] {
    justify-content: space-between;
  }
  .sm\\\\:h-34[data-v-0f84999b] {
    height: 8.5rem;
  }
  .sm\\\\:text-base[data-v-0f84999b] {
    font-size: 1rem;
    line-height: 1.5rem;
  }
  .sm\\\\:text-5xl[data-v-0f84999b] {
    font-size: 3rem;
    line-height: 1;
  }
  .sm\\\\:min-h-\\\\[220px\\\\][data-v-0f84999b] {
    min-height: 220px;
  }
  .sm\\\\:px-6[data-v-0f84999b] {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
  .sm\\\\:px-28[data-v-0f84999b] {
    padding-left: 7rem;
    padding-right: 7rem;
  }
  .sm\\\\:col-span-1[data-v-0f84999b] {
    -ms-grid-column-span: span 1 / span 1;
    grid-column: span 1 / span 1;
  }
}
@media (min-width: 768px) {
  .md\\\\:min-h-\\\\[180px\\\\][data-v-0f84999b] {
    min-height: 180px;
  }
}
@media (min-width: 1024px) {
  .lg\\\\:flex-col[data-v-0f84999b] {
    flex-direction: column;
  }
  .lg\\\\:order-none[data-v-0f84999b] {
    order: 0;
  }
  .lg\\\\:min-h-min[data-v-0f84999b] {
    min-height: -moz-min-content;
    min-height: min-content;
  }
  .lg\\\\:py-7[data-v-0f84999b] {
    padding-bottom: 1.75rem;
    padding-top: 1.75rem;
  }
  .lg\\\\:px-8[data-v-0f84999b] {
    padding-left: 2rem;
    padding-right: 2rem;
  }
  .lg\\\\:grid-cols-10[data-v-0f84999b] {
    grid-template-columns: repeat(10, minmax(0, 1fr));
  }
  .lg\\\\:col-span-10[data-v-0f84999b] {
    -ms-grid-column-span: span 10 / span 10;
    grid-column: span 10 / span 10;
  }
  .lg\\\\:col-span-6[data-v-0f84999b] {
    -ms-grid-column-span: span 6 / span 6;
    grid-column: span 6 / span 6;
  }
  .lg\\\\:col-span-4[data-v-0f84999b] {
    -ms-grid-column-span: span 4 / span 4;
    grid-column: span 4 / span 4;
  }
}
@media (prefers-color-scheme: dark) {
  .dark\\\\:bg-black[data-v-0f84999b] {
    --tw-bg-opacity: 1;
    background-color: rgba(0, 0, 0, var(--tw-bg-opacity));
  }
  .dark\\\\:bg-white\\\\/10[data-v-0f84999b] {
    --tw-bg-opacity: 0.1;
    background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
  }
  .dark\\\\:bg-gray-900[data-v-0f84999b] {
    --tw-bg-opacity: 1;
    background-color: rgba(24, 24, 27, var(--tw-bg-opacity));
  }
  .dark\\\\:border-transparent[data-v-0f84999b] {
    border-color: transparent;
  }
  .dark\\\\:border-gray-900[data-v-0f84999b] {
    --tw-border-opacity: 1;
    border-color: rgba(24, 24, 27, var(--tw-border-opacity));
  }
  .dark\\\\:border-none[data-v-0f84999b] {
    border-style: none;
  }
  .dark\\\\:block[data-v-0f84999b] {
    display: block;
  }
  .dark\\\\:hidden[data-v-0f84999b] {
    display: none;
  }
  .dark\\\\:text-white[data-v-0f84999b] {
    --tw-text-opacity: 1;
    color: rgba(255, 255, 255, var(--tw-text-opacity));
  }
  .dark\\\\:text-gray-300[data-v-0f84999b] {
    --tw-text-opacity: 1;
    color: rgba(189, 189, 189, var(--tw-text-opacity));
  }
  .dark\\\\:hover\\\\:text-white[data-v-0f84999b]:hover {
    --tw-text-opacity: 1;
    color: rgba(255, 255, 255, var(--tw-text-opacity));
  }
}
"
`;
